airflow/contrib/operators/kubernetes_operator.py

```diff
@@ -0,0 +1,129 @@
+from airflow.models import BaseOperator
+import jinja2
+import logging
+import re
+import subprocess
+import tempfile
+import time
+import uuid
+
+YAML_TEMPLATE = """
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ job_name }}
+spec:
+  template:
+    spec:
+      containers: {% for container in containers %}
+      - name: {{ container.name }}
+        image: {{ container.image }}
+        command: {{ container.command }} {% endfor %}
+      restartPolicy: Never
+  backoffLimit: 0
+"""
+
+
+def generate_yaml(unique_job_name, kubernetes_container_information_list):
+    """
+    Generate yaml string from YAML_TEMPLATE and inputs.
+    """
+    template = jinja2.Template(YAML_TEMPLATE)
+    return template.render(job_name=unique_job_name, containers=kubernetes_container_information_list)
+
+
+class KubernetesContainerInformation():
+    """
+    Information for an individual container,
+    used to generate Kubernetes Job yamls.
+    """
+    def __init__(self,
+                 name,
+                 image,
+                 command):
+        self.name = name
+        self.image = image
+        self.command = command
+
+# Amount of time to sleep between polling for the job status
+SLEEP_TIME_BETWEEN_POLLING = 60
+
+
+class KubernetesJobOperator(BaseOperator):
+    """
+    The KubernetesJobOperator spins up, executes, and cleans up a Kubernetes job.
+    It does so by following these steps:
+        1. Create a Kubernetes Job yaml from job_name and kubernetes_container_information_list
+        2. Instatiate the Job
+        3. Poll and wait for Job completion
+        4. Delete the Job (and Pod)
+
+    # TODO: Expand yaml creation or allow people to pass custom Kubernetes Job yamls.
+            That way we can specify things like: resources, env_vars, volumes, etc.
+
+    :param job_name: meaningful name for the job
+    :type job_name: string
+    :kubernetes_container_information_list: list of containers required to execute the Job
+    :type kubernetes_container_information_list: list of type KubernetesContainerInformation
+    """
+    def __init__(self,
```

Comment:
Quick idea @JLDLaughlin -- Do we wanna create a mailing list `airflow-status@bluecore.com` and default this to always send job status updates to that email? Interested folks can join the list or something. Otherwise folks can set more specific email addresses for their job statuses, but at the very least we have all jobs sending status updates somewhere. I feel like that may be useful, not very sure how. Also not a blocker here.